### PR TITLE
Bibtexextractor

### PR DESCRIPTION
**Pull request by borovikovnik**
Add parser for extraction BibTeX fields from plain text
Available for parsing authors, title, journal, publisher, year, pages, urls

### CODE REVIEW

**Comment by Siedlerchr** [comment]
Thanks for your contribution! Have you seen the issue #4826 ?

**Comment by koppor** [comment]
@Siedlerchr There is a quick custom algorithm included. I would go with that as a first step. - In case the JavaFX code is good enough, I would merge and then fix the codestyle in a separate PR.

File: src/main/java/org/jabref/gui/bibtexextractor/ExtractBibtexDialog.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+/**
+ * GUI Dialog for the feature "Extract BibTeX from plain text".
+ */
+public class ExtractBibtexDialog extends BaseDialog<Void> {
+
+    private final JabRefFrame frame;
+    private TextArea textArea;
+    private Button buttonExtract;
+
+    public ExtractBibtexDialog(JabRefFrame frame) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/bibtexextractor/ExtractBibtexDialog.java]: We don't JabRef Frame here.
Best is to use the StateManager and then get the active Database and call insertEntries there

**Comment by Siedlerchr** [comment]
It would be nice if you could fix the remaining issues.

**Comment by koppor** [comment]
@Siedlerchr just took over at #jabcon [https://jabcon.jabref.org](http://jabcon.jabref.org/).

- For adding online services, see https://github.com/koppor/jabref/issues/315
- Due to migration to JAvaFX, the feature was deleted at https://github.com/JabRef/jabref/pull/4717

**Comment by koppor** [comment]
Follow up at https://github.com/JabRef/jabref/pull/5206.

**Pull request closed without merging**